### ISSUE TITLE
style.css href when using a custom Express server

### ISSUE THREAD

**Issue by colinrcummings** [zeit/next-plugins#56]
I'm using a [custom Express server](https://github.com/zeit/next.js/blob/canary/examples/custom-server-express/server.js) with [next.js in a subdirectory](https://github.com/zeit/next.js/issues/819#issuecomment-308098780). I have installed `@zeit/next-css` and added `next.config.js` and `pages/_document.js` per the [README](https://github.com/zeit/next-plugins/tree/master/packages/next-css); however, `/_next/static/style.css` returns a 404. I've confirmed that the file is being built to `.next/static/style.css`. I've also tried to ping it at a few different URLs (e.g. `/_next/-/static/style.css`), but I can't find where it's being served. This is all true in both development and production. Do I need to configure the `static` folder in my custom `server.js` (code below)?

````
const express = require('express');
const next = require('next');

const dev = process.env.NODE_ENV !== 'production';
const app = next({ dir: './client', dev });
const handle = app.getRequestHandler();

app
  .prepare()
  .then(() => {
    const server = express();

    // custom routes
    server.get('/s/:id', (req, res) => {
      const actualPage = '/show';
      const queryParams = { id: req.params.id };
      app.render(req, res, actualPage, queryParams);
    });

    // default route
    server.get('*', (req, res) => {
      return handle(req, res);
    });

    server.listen(3000, err => {
      if(err) throw err;
      console.log('> Ready on http://localhost:3000');
    });
  })
  .catch(ex => {
    console.error(ex.stack);
    process.exit(1);
  });
````

**Comment by timneutkens** [comment]
Should work without any additions 🤔 https://github.com/zeit/next.js/blob/c92bc858f7665b6ea1a2e2d80455e47faf1aefec/server/index.js#L244

**Comment by gjjimenez** [comment]
I have the same problem, only production.
/_next/static/style.css returns a 404.

I've also tried to ping it at a few different URLs
/src/_next/static/style.css
./src/_next/static/style.css

**Comment by timneutkens** [comment]
Just tried and I can't reproduce based on your code. Can you provide a minimal reproduction 👍

**Comment by colinrcummings** [comment]
@timneutkens, thanks for taking a look. See [here](https://github.com/colinrcummings/next-with-css).

**Comment by timneutkens** [comment]
`const app = next({ dir: './client', dev });`

You're starting your app with `./client`, but the server is inside the client folder. That's the reason.

**Comment by colinrcummings** [comment]
Changing the line to

`const app = next({ dir: '.', dev });` 

or 

`const app = next({ dev });` 

both cause the index page endpoint to throw a 404...

**Comment by colinrcummings** [comment]
I added the following to `server.js`:

`server.use('/static', express.static(path.join(__dirname, '.next/static')));`

and changed the link in `pages\_document.js` to:

`<link rel="stylesheet" href="/static/style.css" />`

which works in development and production. Any potential watchouts with this approach?

**Comment by lfender6445** [comment]
TLDR - this issue was fixed here 

- https://github.com/zeit/next.js/pull/3722
- https://github.com/zeit/next.js/commit/955cc82736f6e1e3913a8d94d549e40e5cf296d0

## my setup
I was running into all of the same issues as @colinrcummings - after a lot of digging around there seems to be a discrepancy between where the `distDir` or dist directory gets place relative to a custom src directory. 

Heres my setup: 

```js
// next.config.js
const withCSS = require('@zeit/next-css')

const opts = {
  distDir: '../dir', // the root of my project
}

module.exports = withCSS(opts)
```

```js
// src/server.js 
// I start my server with `NODE_PATH='./src' node src/server.js`
const next = require('next')

const port = parseInt(process.env.PORT, 10) || 3005
const express = require('express')
const expressRoutes = require('./express-routes')
const bodyParser = require('body-parser')

const dev = process.env.NODE_ENV !== 'production'
const app = next({ dir: 'src', dev, quiet: false })

const runServer = () => {
  const handle = app.getRequestHandler()
  const server = express()

  server.use(bodyParser.json())

  app.prepare()
    .then(() => {
      expressRoutes.register(app, server, handle)

      server.listen(port, (err) => {
        if (err) throw err
        /* eslint-disable no-console */
        console.log(`> Ready on http://localhost:${port}`)
      })
    })
}

runServer()
```

Finally, I had to track down the 404 for the static output - debugging a bunch, I found that the path resolutions inside of `next/server` were not relative to the dist directory. 

https://github.com/zeit/next.js/blob/5.0.0/server/index.js#L243-L246

Changing it to 

```
      '/_next/static/:path*': async (req, res, params) => {
        const p = join(this.dir, this.dist, 'static', ...(params.path || []))
        await this.serveStatic(req, res, p)
      },
```

Did the trick for me, but I'm using a local copy of next. I'm not sure this patch has made it to stable as of today.

**Comment by PaulRBerg** [comment]
Using `^6.0.0-canary.3`and the code is updated as lfender6445 suggested above, however the `style.css` file is still not found.

**Comment by JOnathanST29** [comment]
Try to use "/static/css" 

maybe you are using "static/css"

**Comment by JOnathanST29** [comment]
**Wrong** > `
              <link href="static/css/responsive.css" rel="stylesheet"/>`
**Correct** > `
              <link href="/static/css/responsive.css" rel="stylesheet"/>`
